developer/src/kmanalyze/kmanalyze.cpp

```diff
@@ -154,30 +152,22 @@ BOOL LoadKeyboard(LPSTR fileName, LPKEYBOARD *lpKeyboard)
   if (ckbp->dwFileVersion < VERSION_MIN ||
     ckbp->dwFileVersion > VERSION_MAX)
   {
-    /* Old or new version -- identify the desired program version */
-    if((ckbp->dwFileVersion >= VERSION_160 && ckbp->dwCheckSum == 0) ||
-      /* #7222: We support a zero checksum in Keyman 16.0 and later */
-      VerifyChecksum(buf, &kbp->dwCheckSum, sz))
-    {
-      kbp->dpStoreArray = (LPSTORE)(buf + ckbp->dpStoreArray);
-      for (sp = kbp->dpStoreArray, i = 0; i < kbp->cxStoreArray; i++, sp++)
-        if (sp->dwSystemID == TSS_COMPILEDVERSION)
-        {
-          char buf2[256];
-          wsprintf(buf2, "Wrong File Version: file version is %ls", ((PBYTE)kbp) + (INT_PTR)sp->dpString);
-          delete buf;
-          Err(buf2);
-          return FALSE;
-        }
+    kbp->dpStoreArray = (LPSTORE)(buf + ckbp->dpStoreArray);
+    for (sp = kbp->dpStoreArray, i = 0; i < kbp->cxStoreArray; i++, sp++) {
+      if (sp->dwSystemID == TSS_COMPILEDVERSION)
+      {
+        char buf2[256];
+        wsprintf(buf2, "Wrong File Version: file version is %ls", ((PBYTE)kbp) + (INT_PTR)sp->dpString);
+        delete buf;
+        Err(buf2);
+        return FALSE;
+      }
     }
-    delete buf; Err("Unknown File Version: try using the latest version of KMDECOMP");
+    delete buf;
+    Err("Unknown File Version: try using the latest version of KMDECOMP");
```

Comment:
Hmmm, any guesses where this code came from? 😆

Reply:
yeah, it was looking a little... familiar